joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstCreator.scala

```diff
@@ -50,14 +50,33 @@ class AstCreator(val relativeFileName: String, val parserResult: ParserResult, v
     memberAsts
   }
 
-  protected def astForMembers(members: Seq[DotNetNodeInfo]): Seq[Ast] = members.flatMap(astForNode)
-
+  protected def astForMembers(members: Seq[DotNetNodeInfo]): Seq[Ast] = members.flatMap(member => astForNode(member))
   protected def astForNode(json: Value): Seq[Ast] = {
     val nodeInfo = createDotNetNodeInfo(json)
     astForNode(nodeInfo)
   }
 
-  protected def astForNode(nodeInfo: DotNetNodeInfo): Seq[Ast] = {
+  protected def astForBlock(block: DotNetNodeInfo, order: Int = -1): Ast = {
+    val newBlockNode = blockNode(block).order(order).argumentIndex(order)
+    scope.pushNewScope(BlockScope)
+    var currOrder = 1
+    val childAsts = block
+      .json(ParserKeys.Statements)
+      .arrOpt
+      .getOrElse(List())
+      .arr
+      .flatMap { parserJsonValue =>
+        val parserNode = createDotNetNodeInfo(parserJsonValue)
+        val r          = astForNode(parserNode, currOrder)
+        currOrder = currOrder + r.length
+        r
+      }
+      .toList
+    scope.popScope()
+    blockAst(newBlockNode, childAsts)
+  }
+
+  protected def astForNode(nodeInfo: DotNetNodeInfo, argIndex: Int = -1): Seq[Ast] = {
```

Comment:
Perhaps we should remove the `argIndex` parameter if it's only used in `astForBlock`. In the cases where we use `astForBlock` we can call it directly. In terms of this PR, it is called directly in any case.